src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -776,7 +776,8 @@ impl Step for RustcDev {
         copy_src_dirs(
             builder,
             &builder.src,
-            &["compiler"],
+            // The compiler has a path dependency on proc_macro, so make sure to include it.
+            &["compiler", "library/proc_macro"],
```

Comment:
I've tested that `cargo metadata` and `cargo check` still work in the built rustc-dev tarball. It may be possible to do the same trick for `rustc-literal-escaper` to pull it back into this repo.

Reply:
The rustup component was not the only problem; the other problem are out-of-tree consumers -- RA has to be able to import some of the rustc crates, and IIRC the literal-escaper occurred in the dependency tree.

It'd be good to involve them here as well in case rustc_builtin_macros is in their dependency tree -- Cc @Veykril

Reply:
It is not so that should be fine

Reply:
I am also iffy about the fact that the compiler tree relies on this somewhat magical hack, but it looks fairly simple/maintainable and works perfectly fine. It seems like no one in the project is bothered by it (yet), so I would say let's move forward and revisit it if someone raises an issue.